tempesta_fw/http.c

```diff
@@ -742,6 +743,53 @@ tfw_http_error_resp_switch(TfwHttpReq *req, int status, const char *reason)
 	tfw_http_send_resp(req, code);
 }
 
+static inline void
+tfw_http_srv_hmonitor(TfwHttpResp *resp)
+{
+	u32 crc32;
+	char *body_cstr;
+	TfwServer *srv = (TfwServer *)resp->conn->peer;
+
+	if (!test_bit(TFW_SRV_B_HMONITOR, (unsigned long *)&srv->hm_flags))
+		return;
+
+	if (tfw_apm_hm_srv_limit(resp->status, srv->apmref)) {
+		if (!tfw_srv_suspended(srv)) {
+			tfw_srv_mark_suspended(srv);
+			TFW_WARN_ADDR("server has been suspended: limit"
+				      " for bad responses is exceeded",
+				      &srv->addr);
+		}
+		return;
+	}
+
+	if (!tfw_srv_suspended(srv))
+		return;
+
+	if (!tfw_apm_hm_srv_status_alive(resp->status, srv->apmref)) {
+		body_cstr = tfw_pool_alloc(resp->pool, resp->body.len + 1);
+		if (unlikely(!body_cstr)) {
+			TFW_ERR("Can't allocate memory for hmonitor"
+				" crc32 verification\n");
+			return;
+		}
+		tfw_str_to_cstr(&resp->body, body_cstr, resp->body.len + 1);
+		crc32 = crc32(0, body_cstr, resp->body.len);
+		tfw_pool_free(resp->pool, body_cstr, resp->body.len + 1);
+		if (!tfw_apm_hm_srv_crc32_alive(crc32, srv->apmref))
+			return;
+	}
+
+	tfw_srv_mark_alive(srv);
+}
+
+static inline void
+tfw_http_srv_hm_update(TfwServer *srv, TfwHttpReq *req)
```

Comment:
This comment is for the whole `http.c` module. The new function names are very diverse and it's difficult to attribute a function to new health monitoring functionality. Why not make them have a common prefix, like `tfw_http_hm_XXX`?

Forexample, this function would be named `tfw_http_hm_srv_update()`.

Reply:
Corrected.